### DISCUSSION TITLE
Power timing control

### DISCUSSION THREAD

**Discussion by dwkmff** [jankae/LibreVNA#245] [Ideas]
Can I use Python to control the timing of power for LibreVNA? For example, I want to output -10dbm for 10 seconds and then output -5dbm for 10 seconds, and so on.

**Comment by jankae** [comment]
It depends on what exactly you want to do.

If you just want to generate a signal: Yes, you can use the generator mode for that and can set the output power through the SCPI server (the timing needs to come from your python script). For accurate output amplitudes, you will need to run a source calibration first (refer to the manual). Otherwise the output amplitude may be off by several dBs (especially towards the higher frequencies)

If you want to measure S-parameters with varying stimulus power: you can sort of do that. You can set the stimulus power through SCPI but it can not change mid-sweep. The sweep will restart whenever you change a setting like that.

**Comment by dwkmff** [comment]
Thank you very much for your prompt reply. 
Yes, I just want to generate the signal and time it, and I appreciate your approval of that idea.
